File: dotnet/src/Functions/RotateKey.cs
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Azure.WebJobs;
using Microsoft.Azure.WebJobs.Extensions.Http;
using Microsoft.Extensions.Logging;
using MongoEncryption.Util;
using System;
using System.Threading.Tasks;

namespace MongoEncryption.Functions
{
    public static class RotateKey
    {
        [FunctionName("RotateKey")]
        public static async Task<IActionResult> RunAsync(
            [HttpTrigger(AuthorizationLevel.Function, "put", Route = "rotatekey/{keyName}")] HttpRequest req,
            string keyName,
            ILogger log)
        {
            try
            {
                var mongoCrypt = new MongoCrypt(null, keyName);

                var success = await mongoCrypt.RotateKeyAsync();

                log.LogInformation($"Key {keyName} rotated. Succeeded: {success}");

                return new AcceptedResult();
            }
            catch(Exception ex)
            {
                log.LogError(ex, "Key rotation error");
                return new BadRequestResult();
            }
        }
    }
}

File: java/src/main/java/com/function/RotateKey.java
package com.function;

import java.util.*;
import com.microsoft.azure.functions.annotation.*;
import com.function.util.MongoCrypt;
import com.microsoft.azure.functions.*;

public class RotateKey {

    @FunctionName("RotateKey")
    public HttpResponseMessage run(
            @HttpTrigger(name = "req", 
                methods = {HttpMethod.PUT}, 
                authLevel = AuthorizationLevel.FUNCTION,
                route = "rotatekey/{keyName}") 
            HttpRequestMessage<Optional<String>> request,
            @BindingName("keyName") String keyName,
            final ExecutionContext context) {
        
        try
        {
            var mongoCrypt = new MongoCrypt(null, keyName);

            Boolean success = mongoCrypt.RotateKey();

            context.getLogger().info("Key "+ keyName + " rotated. Succeeded': " + success);

            return request.createResponseBuilder(HttpStatus.ACCEPTED).build();
        }
        catch (Exception ex)
        {
            context.getLogger().warning("Key rotation error. Error: " + ex.getMessage());
            return request.createResponseBuilder(HttpStatus.BAD_REQUEST).build();
        }


    }
}
